app/services/etherdelta_observer.py

```diff
@@ -111,12 +114,12 @@ def validate_order(order, current_block=None):
         try:
             token = market_queue.get()
```

Comment:
`.get()` is a blocking call. It never actually threw `QueueEmpty` in the first place. We can use [`.get_nowait`](https://docs.python.org/3/library/queue.html#queue.Queue.get_nowait) here instead.